kwave/options/simulation_execution_options.py

```diff
@@ -0,0 +1,147 @@
+import os
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Optional, Union
+from warnings import warn
+
+from kwave.ksensor import kSensor
+from kwave.utils.checks import is_unix
+
+
+@dataclass
+class SimulationExecutionOptions:
+    # Are we going to run the simulation on the GPU?
+    # Affects binary name and the way the simulation is run
+    is_gpu_simulation: bool = False
+
+    # user defined location for the binary
+    binary_path: Optional[str] = os.getenv('KWAVE_BINARY_PATH')
+    # user defined location for the binary
+    binary_name: Optional[str] = None
+    # user defined number of threads
+    kwave_function_name: Optional[str] = 'kspaceFirstOrder3D'
+    # Whether to delete the input and output files after the simulation
+    delete_data: bool = True
+    # GPU device flag
+    device_num: Optional[int] = None
+    # number of threads
+    num_threads: Union[int, str] = 'all'
+
+    # user defined thread binding option
+    thread_binding: Optional[bool] = None
+
+    # user input for system string
+    system_call: Optional[str] = None
+    verbose_level: int = 0
+
+    def __post_init__(self):
+        self.validate()
+
+        if (self.binary_path is not None) and (not self.binary_path.endswith(os.path.sep)):
+            self.binary_path = self.binary_path + os.path.sep
+
+        if self.binary_name is None:
+            if self.is_gpu_simulation:
+                self.binary_name = 'kspaceFirstOrder-CUDA' if is_unix() else 'kspaceFirstOrder-CUDA.exe'
+            else:
+                self.binary_name = 'kspaceFirstOrder-OMP' if is_unix() else 'kspaceFirstOrder-OMP.exe'
+
+    def validate(self):
+
+        # check the binary exists and is in the correct place before doing anything else
+        # TODO re-enable this check
+        # if not Path(self.binary_path, self.binary_name).exists():
+        #     warn(f'''
+        #                 The binary file {self.binary_name} could not be found in {self.binary_path}.
+        #                 To use the C++ code, the C++ binaries for your operating system must be downloaded
+        #                 from www.k-wave.org/download.php and placed in the binaries folder.'''
+        #          )
+
```

Comment:
Can this be reenabled?

Reply:
AFAIU we now require binaries to be in a specific path and do not support overriding the path. So, this check is useless actually. I'm happy to remove it if you agree?

Reply:
I agree that changing the binary is an advanced use case, and we can drop support for now. This can be deleted, then.